addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -146,6 +148,14 @@ patch(Chatter.prototype, {
             },
             () => [this.state.thread?.status, this.attachments]
         );
+        useEffect(
+            () => {
+                if (this.state.thread && this.state.highlightMessageId) {
+                    this.highlightMessage();
+                }
+            },
+            () => [this.state.highlightMessageId, this.state.thread]
+        );
```

Comment:
This feature should not be duplicated in `Chatter` when this is also in `Thread` component.
Chatter uses a `Thread`, feature should be implemented once in `Thread`.